src/Patterns/ProductUpdater.php

```diff
@@ -0,0 +1,361 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\Patterns;
+
+/**
+ * Pattern Images class.
+ */
+class ProductUpdater {
+
+	/**
+	 * Generate AI content and assign AI-managed images to Products.
+	 *
+	 * @param array  $vertical_images The vertical images.
+	 * @param string $business_description The business description.
+	 *
+	 * @return bool|\WP_Error True if the content was generated successfully, WP_Error otherwise.
+	 */
+	public function generate_content( $vertical_images, $business_description ) {
+		$last_business_description = get_option( 'last_business_description_with_ai_content_generated' );
+
+		if ( $last_business_description === $business_description ) {
```

Comment:
Do we need to do a check if `$business_description` has a value? On first run if it doesn't then this will evaluate as `false` since `$last_business_description = false` and `$business_description = ""`. Then further down `update_option` will run and create an empty option?

Reply:
Good point: updated https://github.com/woocommerce/woocommerce-blocks/pull/11155/commits/6333f15ef8fd96d0d9e9d0bfa11dd0af2f0d0333.